js/modules/k6/metrics/metrics.go

```diff
@@ -128,14 +128,19 @@ func (m Metric) add(v goja.Value, addTags ...map[string]string) (bool, error) {
 		return raiseNan()
 	}
 
-	tags := state.CloneTags()
+	tags := state.Tags.BranchOut()
 	for _, ts := range addTags {
 		for k, v := range ts {
-			tags[k] = v
+			tags.AddTag(k, v)
```

Comment:
Nit: this will be bad with a *large* amount of `addTags` which arguablt is *really* rare -  have only seen being used with a single tag ever.

A potential fix here will be to get the original js object which has the keys ordered the way they were provided and use that order to only walk the same path in the atlas.

Again I have only seen this being used with 1 tag and ... I would not expect anything noticeable before we are above at least 5-6 tags, probably more like 10+  both of which seem really unlikely to me

Reply:
Fixed in https://github.com/grafana/k6/pull/2632 and rebased here.